pkg/backup/restore_online.go

```diff
@@ -456,8 +459,35 @@ func sendRemoteAddSSTable(
 		batchTimestamp = execCtx.ExecCfg().DB.Clock().Now()
 	}
 
+	// check if file uri is an external connection and swap it out/revalidate
+	locator := file.Dir.URI
+	uri, err := url.Parse(file.Dir.URI)
+	if err != nil {
+		return err
+	}
+	if uri.Scheme == externalconn.Scheme {
+		if err := execCtx.ExecCfg().InternalDB.Txn(ctx, func(ctx context.Context, t isql.Txn) error {
+			ec, err := externalconn.LoadExternalConnection(ctx, uri.Host, t)
+			if err != nil {
+				return err
+			}
+
+			materialized, err := externalconn.Materialize(ec, uri)
```

Comment:
While I'm not worried about materializing the string every time, I do think we need to avoid retrieving the externalconn from the database every time we send a link sst request. Loading the external connection requires an RPC to the external connection table.